cellprofiler/modules/measureradialentropy.py

```diff
@@ -0,0 +1,205 @@
+
+
+
+class MeasurementTemplate(cpm.Module):
+
+    module_name = "MeasureRadialEntropy"
+    category = "Measurement"
+    variable_revision_number = 1
+
+
+    def create_settings(self):
+
+        self.input_object_name = cps.ObjectNameSubscriber(
+            "Select objects to measure",
+            doc="""Select the objects whose radial entropy you want to measure.""" % globals())
+
+        self.input_image_name = cps.ImageNameSubscriber(
+            "Select an image to measure", doc="""Select the
+            grayscale image you want to measure the entropy of.""" % globals())
+
+        self.bin_number=cps.Integer(
+            "Input number of bins", 6, minval=3, maxval=60,
+            doc="""Number of radial bins to divide your object into.  The minimum number
+            of bins allowed is 3, the maximum number is 60.""" % globals())
+
+        self.intensity_measurement=cps.Choice(
+            "Which intensity measurement should be used?", ['Mean','Median'], value='Mean',doc="""
+            Whether each wedge's mean or median intensity should be used to calculate the entropy.""" % globals())
+
+    def settings(self):
+        return [self.input_image_name, self.input_object_name,
+                self.intensity_measurement, self.bin_number]
+
+
+    def run(self, workspace):
+        #Import the workspace and the current measurements
+        meas = workspace.measurements
+        assert isinstance(meas, cpmeas.Measurements)
+
+        statistics = [["Entropy"]]
+
+        workspace.display_data.statistics = statistics
+
+        #Import the settings
+        input_image_name = self.input_image_name.value
+        input_object_name = self.input_object_name.value
+        metric = self.intensity_measurement.value
+        bins = self.bin_number.value
+
+        image_set = workspace.image_set
+
+        input_image = image_set.get_image(input_image_name,
+                                          must_be_grayscale=True)
+        #Read out the pixel data
+        pixels = input_image.pixel_data
+
+        object_set = workspace.object_set
+        assert isinstance(object_set, cpo.ObjectSet)
+
+        objects = object_set.get_objects(input_object_name)
+        labels = objects.segmented
+
+        indexes = objects.indices
+        #Calculate the center of the objects- I'm guessing there's a better way to do this but this was already here
+        centers, radius = minimum_enclosing_circle(labels, indexes)
+
+        feature = self.get_measurement_name(input_image_name,metric,bins)
+        #Do the actual calculation
+        entropy=self.slice_and_measure_intensity(pixels,labels,indexes,centers,metric,bins)
+        #Add the measurement back into the workspace
+        meas.add_measurement(input_object_name,feature,entropy)
+
+        emean = np.mean(entropy)
+        statistics.append([feature, emean])
+
+
+    ################################
+    #
+    # DISPLAY
+    #
+    def display(self, workspace, figure=None):
```

Comment:
It looks like the template is outdated. The proper signature is `def display(self, workspace, figure)`. See the image template for a more up-to-date version of this method.